qsirecon/workflows/base.py

```diff
@@ -138,27 +149,35 @@ def init_single_subject_recon_wf(subject_id):
             "Anatomical (T1w) available for recon: %s", available_anatomical_data
         )
 
+    aggregate_anatomical_nodes = pe.Node(
+        niu.Merge(len(dwi_recon_inputs)),
+        name="aggregate_anatomical_nodes",
+    )
+
     # create a processing pipeline for the dwis in each session
     dwi_recon_wfs = {}
     dwi_individual_anatomical_wfs = {}
     recon_full_inputs = {}
     dwi_ingress_nodes = {}
     anat_ingress_nodes = {}
     print(dwi_recon_inputs)
-    for dwi_input in dwi_recon_inputs:
+    dwi_files = [dwi_input["bids_dwi_file"] for dwi_input in dwi_recon_inputs]
+    for i_run, dwi_input in enumerate(dwi_recon_inputs):
```

Comment:
I have no real idea how `dwi_recon_inputs` organizes the input data, since I assumed that QSIRecon concatenated all of the DWI runs, but since it does collect potentially multiple files, I needed a step to merge them into a single list for the summary reportlet.

Reply:
ohh, in the case of dmri all the concatenation has happened in qsiprep. For qsirecon I think users will want to generate separate results for every dwi file that is in the inputs. If a researcher leaves the files split they most likely want to run recon workflows on them separately to test methoes (eg @ameliecr's project will do this)

Reply:
That makes sense